bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/SmartThingsApiService.java

```diff
@@ -0,0 +1,981 @@
+            return false;
+        }
+        Optional<String> response = Optional.empty();
+        try {
+            String contentString = new Gson().toJson(new JSONContent(capability, command, value));
+            logger.trace("{}: content: {}", host, contentString);
+            InputStream content = new ByteArrayInputStream(contentString.getBytes());
+            String api = API_ENDPOINT_V1 + "devices/" + deviceId + COMMAND;
+            URI uri = new URI("https", null, SMARTTHINGS_URL, 443, api, null, null);
+            response = sendUrl("POST", uri, content);
+        } catch (JsonSyntaxException | URISyntaxException | IOException e) {
+            logger.debug("{}: Send Command to Smartthings Cloud failed: {}", host, e.getMessage());
+        }
+        return response.map(r -> r.contains("ACCEPTED") || r.contains("COMPLETED")).orElse(false);
+    }
+
+    /**
+     * Smartthings API Subscription
+     *
+     *
+     * @return Optional<STSubscription> stSub
+     */
+    public synchronized Optional<STSubscription> smartthingsSubscription() {
+        if (apiKey.isBlank() || deviceId.isBlank()) {
+            return Optional.empty();
+        }
+        Optional<STSubscription> stSub = Optional.empty();
+        try {
+            logger.info("{}: SSE Creating Smartthings Subscription", host);
+            String contentString = new Gson().toJson(new JSONSubscriptionFilter(deviceId));
+            logger.trace("{}: subscription: {}", host, contentString);
+            InputStream subscriptionFilter = new ByteArrayInputStream(contentString.getBytes());
+            URI uri = new URI("https", null, SMARTTHINGS_URL, 443, "/subscriptions", null, null);
+            Optional<String> response = sendUrl("POST", uri, subscriptionFilter);
+            stSub = response.map(r -> new Gson().fromJson(r, STSubscription.class));
+            if (!stSub.isPresent()) {
+                throw new IOException("No Data - is DeviceID correct?");
+            }
+        } catch (JsonSyntaxException | URISyntaxException | IOException e) {
+            logger.warn("{}: SSE Subscription to Smartthings Cloud failed: {}", host, e.getMessage());
+        }
+        return stSub;
+    }
+
+    public synchronized void startSSE() {
+        if (!subscriptionRunning) {
+            logger.trace("{}: SSE Starting job", host);
+            subscription = smartthingsSubscription();
+            logger.trace("{}: SSE got subscription ID: {}", host,
+                    subscription.map(a -> a.getSubscriptionId()).orElse("None"));
+            if (!subscription.map(a -> a.getSubscriptionId()).orElse("").isBlank()) {
+                receiveSSEEvents();
+            }
+        }
+    }
+
+    public void stopSSE() {
+        handlerWrapper.ifPresent(a -> {
+            a.cancel();
+            logger.trace("{}: SSE Stopping job", host);
+            handlerWrapper = Optional.empty();
+            subscriptionRunning = false;
+        });
```

Comment:
3-in-1 suggetion :-) 
- Method content could be moved to the `stop` method. 
- And the Optional is not needed.

```suggestion
        BodyHandlerWrapper handlerWrapper = this.handlerWrapper;
        if (handlerWrapper != null) {
            handlerWrapper.cancel();
            logger.trace("{}: SSE Stopping job", host);
            this.handlerWrapper = null;
            subscriptionRunning = false;
        });
```
- Very minor, if i'm correct you can remove `subscriptionRunning = false;` as that would be the same as `handlerWrapper == null`. Not saying you need to change it, but checking the handlerWrapper is the most reliable and close to the source.